### PR TITLE
[master] Fix BlackDuck PVC Issues: Solr and "-"

### PR DESCRIPTION
**Pull request by operator-r2d2**
This is an automated cherry-pick of #1081

/assign yashbhutwala

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1091#" title="Author self-approved">operator-r2d2</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **yashbhutwala**
You can assign the PR to them by writing `/assign @yashbhutwala` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/blackduck/latest/deployer.go [hunk 1]
```diff
@@ -156,7 +156,7 @@ func (hc *Creater) GetComponents(blackduck *blackduckapi.Blackduck) (*api.Compon
 		componentList.Services = append(componentList.Services, containerCreater.GetDocumentationService())
 	}
 
-	// solr
+	// TODO: solr is not supported in latest (leaving here in case we consolidate the deployers)
```
Comment (msenmurugan) [inline pkg/apps/blackduck/latest/deployer.go]: Is solr not supported in latest?

File: pkg/apps/blackduck/v1/containers/imageTags.go [hunk 2]
```diff
@@ -34,7 +34,6 @@ var imageTags = map[string]map[string]string{
 		"blackduck-cfssl":          "1.0.0",
 		"blackduck-logstash":       "1.0.2",
 		"blackduck-nginx":          "1.0.0",
-		"blackduck-solr":           "1.0.0",
```
Comment (msenmurugan) [inline pkg/apps/blackduck/v1/containers/imageTags.go]: @mphammer why solr is removed in v1?

File: pkg/apps/blackduck/v1/deployer.go [hunk 3]
```diff
@@ -150,7 +150,7 @@ func (hc *Creater) GetComponents(blackduck *blackduckapi.Blackduck) (*api.Compon
 		componentList.Services = append(componentList.Services, containerCreater.GetDocumentationService())
 	}
 
-	// solr
+	// TODO: solr is not supported in v1 (leaving here in case we consolidate the deployers)
```
Comment (msenmurugan) [inline pkg/apps/blackduck/v1/deployer.go]: v1 needs solr

**Comment by yashbhutwala** [comment]
@mphammer creating another cherry pick

**Pull request closed without merging**